test/ILLink.Tasks.Tests/ILLink.Tasks.Tests.cs

```diff
@@ -106,20 +113,21 @@ public void TestReferenceAssemblyPaths (string[] referenceAssemblyPaths)
 				var actualReferences = driver.GetReferenceAssemblies ();
 				Assert.Equal (expectedReferences.OrderBy (a => a), actualReferences.OrderBy (a => a));
 				foreach (var reference in expectedReferences) {
-					var referenceName = Path.GetFileNameWithoutExtension (reference);
-					var actualAction = driver.Context.Actions[referenceName];
+					var ad = new Mono.Cecil.AssemblyNameDefinition (Path.GetFileNameWithoutExtension (reference), new Version ());
+					AssemblyAction actualAction = driver.Context.CalculateAssemblyAction (ad);
 					Assert.Equal (AssemblyAction.Skip, actualAction);
 				}
 			}
 		}
 
 		[Theory]
-		[InlineData (new object[] { new string[] { "AssemblyName" } })]
-		public void TestRootAssemblyNames (string[] rootAssemblyNames)
+		[InlineData (new object[] { new string[] { "illink.dll" } })]
```

Comment:
Could you keep the test without extension? That's really the expected use for RootAssemblyNames.

Reply:
It's not how it's used today. This test change actually reflects how it's used today.

Reply:
The SDK uses this for [TrimmerRootAssembly](https://docs.microsoft.com/en-us/dotnet/core/deploying/trimming-options#root-assemblies) which expects no extension.

Reply:
Oh boy, made this quirk to work as well

Reply:
It's not a quirk, it's documented behavior: "Additional assemblies may be "rooted" by name (without the .dll extension):"

Ref: https://docs.microsoft.com/en-us/dotnet/core/deploying/trimming-options#root-assemblies